### PR TITLE
Use check in params so we don't ignore errors

### PR DESCRIPTION
**Pull request by vdemeester**
Small improvements mainly on the `TearDownTest` for the suite(s) so we make sure to not ignore some errors.

/cc @cpuguy83 @dnephin @icecrime @thaJeztah @AkihiroSuda @vieux @tiborvass 

🐻

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: integration-cli/docker_utils.go [hunk 1]
```diff
@@ -303,26 +292,26 @@ func getAllNetworks() ([]types.NetworkResource, error) {
...
+		}
+		status, b, err := sockRequest("DELETE", "/plugins/"+pluginName+":"+tag+"?force=1", nil)
 		if err != nil {
-			errors = append(errors, err.Error())
+			errs = append(errs, err.Error())
 			continue
 		}
-		if status != http.StatusNoContent {
-			errors = append(errors, fmt.Sprintf("error deleting plugin %s: %s", p.Name, string(b)))
+		if status != http.StatusOK {
```
Comment (vdemeester) [inline integration-cli/docker_utils.go]: plugin remove doesn't answer `http.StatusNoContent` (maybe it should but it does not)

File: integration-cli/docker_utils.go [hunk 2]
```diff
@@ -389,21 +371,16 @@ func deleteAllImages() error {
 		fields := strings.Fields(l)
 		imgTag := fields[0] + ":" + fields[1]
 		if _, ok := protectedImages[imgTag]; !ok {
-			if fields[0] == "<none>" {
+			if fields[0] == "<none>" || fields[1] == "<none>" {
```
Comment (vdemeester) [inline integration-cli/docker_utils.go]: This make sure it will also delete image like `imageName:<none>` which are generated on some `DockerRegistrySuite` tests. There was some left overs, and as we ignored the errors, we didn't even know it…

**Comment by vdemeester** [comment]
It showed some errors that weren't catched… updates should fix them :angel:

File: integration-cli/check_test.go [hunk 3]
```diff
@@ -44,12 +44,14 @@ func (s *DockerSuite) OnTimeout(c *check.C) {
...
-	deleteAllImages()
-	deleteAllVolumes()
-	deleteAllNetworks()
-	deleteAllPlugins()
+	unpauseAllContainers(c)
+	deleteAllContainers(c)
+	deleteAllImages(c)
+	deleteAllVolumes(c)
+	deleteAllNetworks(c)
+	if daemonPlatform == "linux" {
```
Comment (vdemeester) [inline integration-cli/check_test.go]: Plugins aren't supported by `windows` (and possibly other daemon platform), so we should try to clean them on those.

File: integration-cli/docker_test_vars.go [hunk 4]
```diff
@@ -98,9 +81,6 @@ func init() {
 		fmt.Printf("ERROR: couldn't resolve full path to the Docker binary (%v)\n", err)
 		os.Exit(1)
 	}
-	if registryImage := os.Getenv("REGISTRY_IMAGE"); registryImage != "" {
```
Comment (thaJeztah) [inline integration-cli/docker_test_vars.go]: Do we know why this was there?

File: integration-cli/docker_utils.go (hunk 1 above)
Comment (thaJeztah) [inline integration-cli/docker_utils.go]: I think we should fix that if it doesn't return content /cc @anusha-ragunathan

File: integration-cli/docker_cli_restart_test.go [hunk 5]
```diff
@@ -275,4 +275,7 @@ func (s *DockerSuite) TestRestartAutoRemoveContainer(c *check.C) {
 
 	out, _ = dockerCmd(c, "ps")
 	c.Assert(out, checker.Contains, id[:12], check.Commentf("container should be restarted instead of removed: %v", out))
+
+	// Kill the container to make sure it will be removed
+	dockerCmd(c, "kill", id)
```
Comment (tonistiigi) [inline integration-cli/docker_cli_restart_test.go]: Why is this needed now?

File: integration-cli/docker_utils.go (hunk 1 above)
Comment (vdemeester) [inline integration-cli/docker_utils.go]: I can add a TODO 👼

File: integration-cli/docker_cli_restart_test.go (hunk 5 above)
Comment (vdemeester) [inline integration-cli/docker_cli_restart_test.go]: So, this tests is making sure a container started with `--rm` will not be removed when doing a restart one it. This means, when `TearDown` gets there, it's failing with `Removal in progress`. By killing it before, I make sure it's getting cleaned before.

File: integration-cli/docker_test_vars.go (hunk 4 above)
Comment (vdemeester) [inline integration-cli/docker_test_vars.go]: Not really, but it's dead code 😅

**Comment by tonistiigi** [comment]
LGTM

File: integration-cli/docker_utils.go [hunk 6]
```diff
@@ -373,37 +357,39 @@ func getAllVolumes() ([]*types.Volume, error) {
 
 var protectedImages = map[string]struct{}{}
 
-func deleteAllImages() error {
-	cmd := exec.Command(dockerBinary, "images")
+func deleteAllImages(c *check.C) {
```
Comment (thaJeztah) [inline integration-cli/docker_utils.go]: As a follow up, we should use `docker image prune -af` :smile:

**Review by thaJeztah — APPROVED** [review]
LGTM
